src/jwt/refresh-token.js

```diff
@@ -0,0 +1,128 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | "_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+
+/**
+ * @file Refresh token service
+ * @description In charge of refreshing the access tokens when is needed it.
+ * @requires jquery
+ * @requires ju-shared/observable-class
+ * @requires ju-shared/jwt/auth-provider
+ * @requires ju-shared/jwt/proxy
+ * @module ju-shared/jwt/refresh-token
+ * @extends ju-shared/observable-class
+ * @listens module:ju-shared/jwt/auth-provider#TOKEN_UPDATED
+ */
+
+define([
+        "jquery",
+        "ju-shared/observable-class",
+        "ju-shared/jwt/auth-provider",
+        "ju-shared/jwt/proxy"
+    ],
+    function (
+        $,
+        ObservableClass,
+        AuthProvider,
+        AuthProxy
+    ){
+        "use strict";
+
+
+        /** Constants **/
+
+        /**
+         * @constant {Number} NEEDED_TIME_TO_REFRESH - time to refresh the token before it expires
+         */
+        var NEEDED_TIME_TO_REFRESH = 2 * 60;
+
+
+        // timeout id
+        var timeoutId = null;
+
+
+        var RefreshToken = ObservableClass.extend({
+
+            /**
+             * @constructor
+             * @alias module:ju-shared/jwt/refresh-token
+             * @param {Boolean} [autoStart] - indicates if want to start the timeout on the init phase
+             */
+            init : function (autoStart) {
+                this.proxy = AuthProxy.getInst();
+                this.authProvider = AuthProvider.getInst();
+                this.authProvider.on(AuthProvider.EV.TOKEN_UPDATED, $.proxy(this.restart,this));
+                if (autoStart){
```

Comment:
can we remove this flag?

I prefer `start` to be manually called after the initialization as side effects in constructors are a bad practice.